common/src/main/java/uk/gov/gchq/palisade/policy/Rules.java

```diff
@@ -154,30 +155,43 @@ public String getMessage() {
 
     @Override
     public boolean equals(final Object o) {
-        if (this == o) {
-            return true;
+        boolean rtn = (this == o);
+        if (!rtn) {
+            if (o != null && this.getClass() == o.getClass()) {
+
+                final Rules<?> that = (Rules<?>) o;
+
+                final Set<String> thisKeySet = this.rules.keySet();
+                final Set<String> thatKeySet = that.getRules().keySet();
+                final EqualsBuilder builder = new EqualsBuilder()
+                        .append(message, that.message)
+                        .append(thisKeySet, thatKeySet);
+
+                if (builder.isEquals()) {
+                    final String[] ruleNames = thisKeySet.toArray(new String[thisKeySet.size()]);
+                    for (int i = 0; i < this.rules.size() && builder.isEquals(); i++) {
```

Comment:
it would be more efficient to iterate over the entry set.

Reply:
I loose the neat `builder.isEquals()` in the loop termination.

Reply:
@GCHQDeveloper404 - I've pushed up the change I was suggesting. Let me know what you think.